accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,298 @@
+  2. AutoYaST deployment: in case the user wants to deploy additional autoinstalled node, SUSE Manager should guide the user into setting up as many nodes of the cluster by autoinstalling them using the AutoYaST profile provided by `skuba`.
+
+  **NOTE**: skuba also provides templates for AWS and libvirt, but those are not supported at the moment of writing.
+
+  ##### The Management Node
+
+  CaaS Platform requires a "Management Node" to deploy itself. The Management node must run SUSE Linux Enterprise 15 SP1 or an OpenSUSE equivalent (TODO: check with CaaSP team: there aren't repos for OpenSUSE).
+  The Management Node will be used for deployment of the CaaS Platform cluster, and cannot be re-used to be part of the cluster.
+
+  Provided that the machine is a minion (Salt-only feature) and is running the correct OS (checked via grains), we can introduce a new entitlement ("Add-On System type") to entitle the machine as a Management Node.
+
+  As it already happens with Container Build Host and OS Image Build Host, the user has also to assign the following channels to the Management Node:
+
+  - SUSE CaaS Platform Extension to SUSE Linux Enterprise 15
+  - SUSE Containers Module 15 SP1 x86_64 
+
+  These channels must be already synced by SUSE Manager.
+
+  In the corresponding Salt state, that must be applied via the usual highstate, the state must:
+
+  - Generate an SSH keypair for accessing the cluster nodes. It is also possible to generate the keys locally to the SUSE Manager server and make use of SSH's forwarding agent mechanism.
+  - Install the pattern `SUSE-CaaSP-Management` (depends on `terraform` and `skuba` packages)
+
+  e.g.
+
+  ```
+  mgr_install_caasp_tools:
+  pkg.installed:
+    - pkgs:
+      - SUSE-CaaSP-Management
+
+  mgr_sshd_installed:
+    pkg.installed:
+      - name: openssh
+  
+  generate_ssh_key_management:
+    cmd.run:
+      - name: ssh-keygen -q -N '' -f /var/lib/caasp-management/.ssh/id_rsa
+      - unless: test -f /var/lib/caasp-management/.ssh/id_rsa
+    require:
+      - pkg: mgr_sshd_installed
+  ```
+
+  Finally, SUSE Manager copies the templates for deployment files provided by `skuba` (in `/usr/share/caasp/AutoYaST/bare-metal/AutoYaST.xml`) from the Management Node to the SUSE Manager itself.
+
+  Given that the Management Node is a Salt minion managed by SUSE Manager, every update for `SUSE-CaaSP-Management` can be installed directly from SUSE Manager. Extra care should be put into protecting the cluster configuration file generated by `skuba` during cluster bootstrapping (rsync into SUSE Manager nightly?).
+
+  ##### Load balancer
+
+  Every CaaS Platform production cluster must have at least one load balancer associated. There is not a standardized way to deploy a load balancer. One example that the CaaS Platform suggests using is the [`nginx` TCP Load Balancer with passive checks](https://susedoc.github.io/doc-caasp/beta/caasp-deployment/single-html/#_configuring_the_load_balancer).
```

Comment:
It's important to note that nginx has no active health checks in the open source version; for having active health checks we would need to use `haproxy` (it can work in L4 mode while performing L7 active health checks).

Reply:
Ok, I added the "health" in "passive checks". Should we mention `haproxy` too? Do you have a reference configuration in the deployment guide?